lib/elastic_apm/context_builder.rb

```diff
@@ -42,6 +42,16 @@ def apply_to_request(context, rack_env)
     end
     # rubocop:enable Metrics/MethodLength, Metrics/AbcSize
 
+    def should_capture_body?(for_type)
+      option = config.capture_body
+
+      return true if option == 'all'
+      return true if option == 'transactions' && for_type == :transaction
+      return true if option == 'errors' && for_type == :error
+
+      false
```

Comment:
The configuration doesn't change between requests, so you could store it as a class variable and then create another class var holding a hash with mappings for the type. Something along the lines of:
```
m = { transaction: true, error: true }
case config.capture_body
when 'off'
  m[:transaction] = false
  m[:error] = false
when 'transaction':
  m[:error] = false
when 'error':
  m[:transaction] = false
```

And then access it within `should_capture_body?(for_type)` with `m[for_type]`

Reply:
Oh, so memoize the method basically?

Reply:
Skipping because the lookups aren't expensive in the first place 😊